### DISCUSSION TITLE
How to add a rich text plugin to handle pasted content?

### DISCUSSION THREAD

**Discussion by EasonSoong** [payloadcms/payload#984] [Q&A]
I found payload rich text editor has a withHTML plugin by default. And it handles the pasted content from other places like google doc. When I try to use my own plugin to handle the pasted content. I found it's impossible because the withHTML's plugin will be executed before my plugin and just return in some conditions without triggering my plugin. Is there a way to solve this problem?

**Comment by EasonSoong** [comment]
Hello guys, any suggestions about this?

**Comment by jmikrut** [comment]
Hey @EasonSoong — this is a good question. I think we need to re-order plugin execution, so that your plugin is executed _**before**_ the built-in paste handling. That will solve this issue. It would be a trivial PR to make as well if you are up for it. Would just need to reorder the plugin wrapping within the rich text component itself.

Are you up for a PR? Otherwise we can get to this at some point soon!

**Reply by EasonSoong** [reply]
Yes, I would love to add this pr. I am using a rich editor plugin to support this feature for now. At the plugin I use an input element to handle the paste event and do the data transformations, and I think paste directly from the editor could improve ux.

![image](https://user-images.githubusercontent.com/57319837/187647230-94649155-1dbd-4302-a3f6-9cb86d70351c.png)
